clients/js/src/generated/instructions/addCanopy.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  ResolvedAccount,
+  ResolvedAccountsWithIndices,
+  expectPublicKey,
+  getAccountMetasAndSigners,
+} from '../shared';
+
+// Accounts.
+export type AddCanopyInstructionAccounts = {
+  treeConfig?: PublicKey | Pda;
+  merkleTree: PublicKey | Pda;
+  treeCreatorOrDelegate?: Signer;
+  logWrapper?: PublicKey | Pda;
+  compressionProgram?: PublicKey | Pda;
+  systemProgram?: PublicKey | Pda;
+};
+
+// Data.
+export type AddCanopyInstructionData = {
+  discriminator: Array<number>;
+  startIndex: number;
+  canopyNodes: Array<Uint8Array>;
+};
+
+export type AddCanopyInstructionDataArgs = {
+  startIndex: number;
+  canopyNodes: Array<Uint8Array>;
+};
+
+export function getAddCanopyInstructionDataSerializer(): Serializer<
+  AddCanopyInstructionDataArgs,
+  AddCanopyInstructionData
+> {
+  return mapSerializer<
+    AddCanopyInstructionDataArgs,
+    any,
+    AddCanopyInstructionData
+  >(
+    struct<AddCanopyInstructionData>(
+      [
+        ['discriminator', array(u8(), { size: 8 })],
+        ['startIndex', u32()],
+        ['canopyNodes', array(bytes({ size: 32 }))],
+      ],
+      { description: 'AddCanopyInstructionData' }
+    ),
+    (value) => ({
+      ...value,
+      discriminator: [247, 118, 145, 92, 84, 66, 207, 25],
+    })
+  ) as Serializer<AddCanopyInstructionDataArgs, AddCanopyInstructionData>;
+}
+
+// Args.
+export type AddCanopyInstructionArgs = AddCanopyInstructionDataArgs;
+
+// Instruction.
+export function addCanopy(
+  context: Pick<Context, 'eddsa' | 'identity' | 'programs'>,
+  input: AddCanopyInstructionAccounts & AddCanopyInstructionArgs
+): TransactionBuilder {
+  // Program ID.
+  const programId = context.programs.getPublicKey(
+    'mplBubblegum',
+    'BGUMAp9Gq7iTEuizy4pqaxsTyUCBK68MDfK752saRPUY'
+  );
+
+  // Accounts.
+  const resolvedAccounts: ResolvedAccountsWithIndices = {
+    treeConfig: {
+      index: 0,
+      isWritable: false,
+      value: input.treeConfig ?? null,
+    },
+    merkleTree: { index: 1, isWritable: true, value: input.merkleTree ?? null },
+    treeCreatorOrDelegate: {
+      index: 2,
+      isWritable: false,
+      value: input.treeCreatorOrDelegate ?? null,
+    },
+    logWrapper: {
+      index: 3,
+      isWritable: false,
+      value: input.logWrapper ?? null,
+    },
+    compressionProgram: {
+      index: 4,
+      isWritable: false,
+      value: input.compressionProgram ?? null,
+    },
+    systemProgram: {
+      index: 5,
+      isWritable: false,
+      value: input.systemProgram ?? null,
+    },
+  };
+
+  // Arguments.
+  const resolvedArgs: AddCanopyInstructionArgs = { ...input };
+
+  // Default values.
+  if (!resolvedAccounts.treeConfig.value) {
+    resolvedAccounts.treeConfig.value = findTreeConfigPda(context, {
+      merkleTree: expectPublicKey(resolvedAccounts.merkleTree.value),
+    });
+  }
+  if (!resolvedAccounts.treeCreatorOrDelegate.value) {
+    resolvedAccounts.treeCreatorOrDelegate.value = context.identity;
+  }
```

Comment:
Calling this `treeCreatorOrDelegate` actually mismatches the behavior of the current instruction which only allows for tree delegate to be used.  However, I made another comment there asking why it cannot be either tree creator or delegate like in other instructions.